client/src/containers/CampaignAdmin/tests/CampaignAdminList-test.js

```diff
@@ -119,4 +119,48 @@ describe('CampaignAdminList', () => {
       expect(cmp.instance().getSelectedItem().Title).toBe('Page two');
     });
   });
+
+  describe('componentDidMount_set_error', () => {
+    let cmp = null;
+
+    it('should return 404 error code if server returns 404', (done) => {
+      const modProps = { ...props };
+      modProps.itemListViewEndpoint = { url: '', method: 'GET' };
+      modProps.recordActions = {
+        fetchRecord: jest.fn(() => new Promise(() => {
+          const error = new Error('Not Found');
+          error.response = { status: 404 };
+          throw error;
+        })),
+      };
+      modProps.record = {};
+
+      cmp = shallow(<CampaignAdminList {...modProps} />);
+      cmp.instance().componentDidMount();
+      setImmediate(() => {
+        expect(cmp.state().errorCode).toBe(404);
```

Comment:
We should probably also check that `error` in the state is set to `true`.

Reply:
DONE